RFC/src/RFC-0230_HTLC.md

```diff
@@ -48,49 +48,75 @@ technological merits of the potential system outlined herein.
+* MUST include the lock height in the signature challenge to stop tampering with the lock height after the transaction was constructed.
+
+This add the following requirement to a miner:
+* MUST not add any transaction to the new [block] to be mined where the blockchain's block height has not already exceeded the transaction's lock height.
+
+This also adds the following requirement to a [base node]:
+* MUST reject any [block] that contains a kernel with a lock height not already past the [current head].
+
+#### Time Locked UTXOs
+
+Time locked UTXOs can be accomplished by modifying the feature flags of a UTXO to include a lock height. This allows a limit as to when in the blockchain lifetime the specific UTXO can be spent. 
+
+This requires that users constructing a transaction:
+
+- MUST include a lock height in the feature flag of their UTXO,
+- MUST include the lock height in the signature challenge to stop tampering with the lock height after the transaction was constructed.
 
 This add the following requirement to a miner:
-* MUST not add any transaction to the mined [block](block) that has not already exceeded its lockheight.
 
-This also adds the following requirement to a [base node](base node):
-* MUST reject any [block](block) that contains a kernel with a lockheight not already past the [currenthead](currenthead).
+- MUST not allow a UTXO to be spent if the blockchain's block height has not already exceeded the UTXO's lock height.
+
+This also adds the following requirement to a [base node]:
+
+- MUST reject any [block] that contains a [UTXO] with a lock height not already past the [current head].
+
+### Mimblewimble N-of-N Multisig UTXO
+
+A normal Mimblewimble UTXO does not have a notion of being a [multisig] UTXO. The UTXO is hidden inside the commitment `C(v,r) = r·G + v·H` by virtue of being blinded. However, the blinding factor `r` can be composed of multiple blinding factors where `r = r1 + r2 + ... + rn`. The output commitment can then be constructed as `C(v,r) = r1·G + r2·G + ... + rn·G + v·H = (r1 + r2 + ... + rn)·G + v·H` where each participant keeps their private blinding factor hidden and only provides their public blinding factor. A multi-party aggregated signature (aggsig) scheme like Musig may be employed to sign such a transaction so that all parties' interests can be protected.
 
-#### Hashed Time Locked Contract
-Unlike Bitcoin where this can be accomplished with a single transaction, in [mimblewimble](MimbleWimble) this is a multi-step process to construct a timelocked contract. 
+The base layer is oblivious as to how the commitment and related signature were constructed. To open such commitment (in order to spend it) only the n-of-n blinding factor `r` is required, and not the original aggsig that was used to sign the transaction. The parties that wants to open the commitment needs to collaborate with special scriptless scripts to produce the n-of-n blinding factor `r`.
```

Comment:
Change "..open such commitment.." to "..open such commitments.."

Reply:
Implemented